### PR TITLE
Setup poetry

### PR DESCRIPTION
**Pull request by nigelmj**
Updated toml file to use poetry instead of setup tools. Created poetry lock file and removed requirements.txt as it is no longer required. Dependencies are now added using `poetry add <package-name>`. Dev specific dependencies can be added using the `--dev` flag before the package name. Ref: #14

### CODE REVIEW

File: pyproject.toml
Comment (suzanneEmbury) [inline pyproject.toml:5]: Thanks for streamlining the syntax here.  Do you know what the convention is for including authors/maintainers in the pyproject.toml file?  Should I be including everyone who has made a contribution through an accepted pull request?  I had a quick search for information but couldn't find anything.

File: pyproject.toml
Comment (suzanneEmbury) [inline pyproject.toml]: Can we set test code dependencies as well?  I have some tests that have not been copied across yet that use a mocking framework that would count as a test dependency.  Since we're leaving an empty dev-dependencies section in case some come up in future, should we have an empty test-dependencies section, too?  Or is that not supported by poetry at present?

**Review by suzanneEmbury — CHANGES_REQUESTED** [review]
This looks good, thanks.  Just a couple of queries to answer - no big changes required.

Overall, though, maybe this PR should include a small change to the README, to give instructions on how to build and publish the package using Poetry?  And also for installing Poetry if you are using the code base as a developer?  I checked out the PR to give it a try and found I did not have Poetry installed on this machine.  It's fairly obvious what the problem is but the developer experience might be a bit smoother if the need to install Poetry first is pointed out?

I installed Poetry (via pipx) and built the egg/wheel.  Then I checked that everything build was being ignored by Git.  It all looks fine.

Thanks again!

**Comment by suzanneEmbury** [comment]
Once this PR is merged, I'll make a new release with Poetry (instead of Twine) as an extra check that all is working well.

File: pyproject.toml
Comment (nigelmj) [reply pyproject.toml:5]: From my understanding, the authors list should only contain the people primarily responsible for the project's creation and ongoing development, not necessarily all contributors. I think they are usually recognised elsewhere, such as in a `CONTRIBUTORS.md` file or in the Contributors section under the Insights tab.

File: pyproject.toml
Comment (nigelmj) [reply pyproject.toml]: Yes! Poetry allows us to declare different dependency groups, so creating a separate section for test dependencies is [fully supported](https://python-poetry.org/docs/managing-dependencies/#:~:text=Poetry%20provides%20a%20way%20to,or%20to%20build%20the%20documentation.). I have updated the `pyproject.toml` file with an empty section for test dependencies to accommodate any future needs.

**Comment by nigelmj** [comment]
The README has been updated with Poetry instructions and a test dependencies section has been added to the `pyproject.toml` file.

File: pyproject.toml
Comment (suzanneEmbury) [reply pyproject.toml:5]: Okay - thanks.  I'll go with that.  Good to know I don't need to remember to add something here every time I merge a PR.

File: pyproject.toml
Comment (suzanneEmbury) [reply pyproject.toml]: Thanks - good to have this here as a reminder.

**Review by suzanneEmbury — APPROVED** [review]
Thanks for making these tweaks.  This PR looks good now.

**Comment by suzanneEmbury** [comment]
Erk ... I fixed the merge conflicts (since I had kept you waiting long enough on this) and now we're getting a CI error.

    E   ModuleNotFoundError: No module named 'loguru'

You've included loguru in the poetry dependencies, so I'm not sure what we've done here.

Are you able to take a further look, @nigelmj ?

**Comment by nigelmj** [comment]
The CI pipeline and the migration to Poetry were done on different branches, which is why the pipeline still relied on the now-deleted requirements.txt. I've now updated the workflow file to run the tests through the virtual environment created with Poetry. Everything is working as expected now!

**Comment by suzanneEmbury** [comment]
Thanks.  The tests also run with poetry in my development environment as well, now, following these same instructions.  So all looks good and I'm going to merge.

**Pull request merged**
